platforms/software/antlr/src/main/java/org/gradle/api/plugins/antlr/internal/AntlrExecuter.java

```diff
@@ -203,8 +203,11 @@ public boolean available() {
     private static class Antlr2Tool extends AntlrTool {
         @Override
         public AntlrResult doProcess(AntlrSpec spec) throws ClassNotFoundException {
-            XRef xref = new MetadataExtracter().extractMetadata(spec.getGrammarFiles());
-            List<GenerationPlan> generationPlans = new GenerationPlanBuilder(spec.getOutputDirectory()).buildGenerationPlans(xref);
+            return doProcess(new GenerationPlanBuilder(spec.getOutputDirectory()).buildGenerationPlans(extractMetadata(spec.getGrammarFiles())), spec);
+        }
+
+        @Nonnull
```

Comment:
imho not needing this we having result or failing for exception instead of passing null shadowing the real cause.